### ISSUE TITLE
No completion for for-loop variable

### ISSUE THREAD

**Issue by winmain** [HaxeFoundation/intellij-haxe#511]
Everytime I use `for`, I don't have completion support for inner variable.

Simple example:

``` haxe
class Test {
  public function test(arr: Array<String>) {
    for (str in arr) {
      st| // <- no completion for `str` here
    }
  }
}
```

Completion popup don't known about `str` variable name in loop body. But when I type `str.`, completion starts to work properly, and I have correct popup list for `str` public functions & vars.

Plugin version: `0.10.1-RC2`
Idea: `2016.2`

**Comment by winmain** [comment]
We found some interesting behavior. Completion **contains** the inner variable with incorrect name:

![Screenshot](https://cloud.githubusercontent.com/assets/167248/19879837/3499354c-a032-11e6-9869-c268228971dd.png)

When I accept this single option, the completion will print `qwe`.

**Comment by bynuff** [comment]
@winmain it will be fixed soon. Maybe in current plugin version.

**Comment by EBatTiVo** [comment]
@bynuff You commented that this will be fixed soon.   Are you working on it?

I believe this is a duplicate of #336, and there is an open PR #452 as a workaround for it.

**Comment by bynuff** [comment]
@EBatTiVo yes, I'm working on it now. This #336 issue already fixed by #456 PR. But current issue not the same, because in this case problem with identifier in loop. #452 not solve the problem. In current code identifiers not includes in suggested variants for autocomplete.

**Comment by EricBishton** [comment]
Was released in December.   My test shows it's fixed.
